simulation/traffic_simulator/src/entity/vehicle_entity.cpp

```diff
@@ -231,7 +244,17 @@ void VehicleEntity::setDecelerationLimit(double deceleration)
     THROW_SEMANTIC_ERROR("Deceleration limit should be over zero.");
   }
   auto behavior_parameter = getBehaviorParameter();
-  behavior_parameter.deceleration = deceleration;
+  behavior_parameter.dynamic_constraints.max_deceleration = deceleration;
+  setBehaviorParameter(behavior_parameter);
+}
+
+void VehicleEntity::setDecelerationRateLimit(double deceleration_rate)
+{
+  if (deceleration_rate <= 0.0) {
+    THROW_SEMANTIC_ERROR("Deceleration limit should be over zero.");
+  }
+  auto behavior_parameter = getBehaviorParameter();
+  behavior_parameter.dynamic_constraints.max_deceleration_rate = deceleration_rate;
```

Comment:
Consider clamping the value to the range 0 to max_deceleration_rate instead of raising an error.
Also, the error message mentions acceleration instead of acceleration rate, which is inappropriate.

Reply:
Deceleration value should not over 0, because it is physically impossible.
So I think raising error is best.

Reply:
**the error message mentions deceleration instead of deceleration rate, which is inappropriate.**